proj: broken on darwin

Somehow it's broken again on Hydra:
https://hydra.nixos.org/build/158008126
```
      Start 48: proj_context_test
48/55 Test #48: proj_context_test ................***Failed    0.03 sec
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from proj_context
[ RUN      ] proj_context.proj_context_set_file_finder
proj_create: Cannot find proj.db
pj_obj_create: Cannot find proj.db
[       OK ] proj_context.proj_context_set_file_finder (1 ms)
[ RUN      ] proj_context.proj_context_set_search_paths
proj_create: Cannot find proj.db
pj_obj_create: Cannot find proj.db
[       OK ] proj_context.proj_context_set_search_paths (0 ms)
[ RUN      ] proj_context.read_grid_from_user_writable_directory
/tmp/nix-build-proj-8.1.1.drv-0/source/test/unit/proj_context_test.cpp:161: Failure
Value of: createTmpFile(filename)
  Actual: false
Expected: true
Cannot open temp_proj_dic3
proj_create: cannot expand +init=temp_proj_dic3:MY_PIPELINE
Cannot open temp_proj_dic3
proj_create: cannot expand +init=temp_proj_dic3:MY_PIPELINE
/tmp/nix-build-proj-8.1.1.drv-0/source/test/unit/proj_context_test.cpp:173: Failure
Expected: (P) != (nullptr), actual: NULL vs (nullptr)
[  FAILED  ] proj_context.read_grid_from_user_writable_directory (9 ms)
[ RUN      ] proj_context.proj_context_set_ca_bundle_path
[       OK ] proj_context.proj_context_set_ca_bundle_path (0 ms)
[----------] 4 tests from proj_context (11 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (11 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] proj_context.read_grid_from_user_writable_directory

 1 FAILED TEST

      Start 49: pj_phi2_test
49/55 Test #49: pj_phi2_test .....................   Passed    0.02 sec
      Start 50: proj_test_cpp_api
50/55 Test #50: proj_test_cpp_api ................   Passed   25.25 sec
      Start 51: gie_self_tests
51/55 Test #51: gie_self_tests ...................   Passed    0.45 sec
      Start 52: test_network
52/55 Test #52: test_network .....................   Passed    0.31 sec
      Start 53: test_defmodel
53/55 Test #53: test_defmodel ....................   Passed    0.05 sec
      Start 54: test_tinshift
54/55 Test #54: test_tinshift ....................   Passed    0.02 sec
      Start 55: test_misc
55/55 Test #55: test_misc ........................   Passed    0.02 sec

98% tests passed, 1 tests failed out of 55

Total Test time (real) =  54.64 sec

The following tests FAILED:
         48 - proj_context_test (Failed)
Errors while running CTest
make: *** [Makefile:136: test] Error 8
builder for '/nix/store/4js5pf04v88ffbg06aznqxschj8284j0-proj-8.1.1.drv' failed with exit code 2
```

--- a/pkgs/development/libraries/proj/default.nix
+++ b/pkgs/development/libraries/proj/default.nix
@@ -42,6 +42,11 @@ stdenv.mkDerivation rec {
     "-DRUN_NETWORK_DEPENDENT_TESTS=OFF"
   ];
 
+  preCheck = ''
+    export HOME=$TMPDIR
+    export TMP=$TMPDIR
+  '';
+
   doCheck = true;
 
   meta = with lib; {

